File: Trees/145.js
// 145. Binary Tree Postorder Traversal
// Given the root of a binary tree, return the postorder traversal of its nodes' values.

var postorderTraversal = function(root) {
    const traveled = []
    if (root) {
        traveled.push(...postorderTraversal(root.left))
        traveled.push(...postorderTraversal(root.right))
        traveled.push(root.val)
    }
    return traveled
};

class TreeNode {
    constructor(val, left, right) {
        this.val = (val === undefined ? 0 : val)
        this.left = (left === undefined ? null : left)
        this.right = (right === undefined ? null : right)
    }
}

const testCase = new TreeNode(1)
testCase.left = new TreeNode(2)
testCase.right = new TreeNode(3)
testCase.left.left = new TreeNode(4)
testCase.left.right = new TreeNode(5)

console.log(postorderTraversal(testCase))

File: Trees/145.py
from typing import Optional, List

class TreeNode:
    def __init__(self, val = 0, left = None, right = None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def postorderTraversal(self, node: Optional[TreeNode]) -> List[int]:
        visited = []
        if not node:
            return visited
        visited += self.postorderTraversal(node.left)
        visited += self.postorderTraversal(node.right)
        visited.append(node.val)
        return visited

test_case = TreeNode(4)
test_case.left = TreeNode(9)
test_case.right = TreeNode(0)
test_case.left.left = TreeNode(5)
test_case.left.right = TreeNode(1)

s = Solution()
print(s.postorderTraversal(test_case))